### PR TITLE
Tracing and tick tracking rework

### PR DESCRIPTION
**Pull request by Jim3Things**

### CODE REVIEW

File: internal/clients/store/storeapi.go [hunk 1]
```diff
@@ -115,95 +115,93 @@ func (store *Store) CreateWithEncode(
...
-			return err
-		}
+	v, err := Encode(m)
 
-		request := &Request{
-			Records:    make(map[string]Record),
-			Conditions: make(map[string]Condition),
-		}
+	if err != nil {
+		return 0, err
```
Comment (MikeCJohnsonUk) [inline internal/clients/store/storeapi.go:133]: You're not going to like me for this, but the symbol for an invalid revision here is "RevisionInvalid".

Occurs lots of times...

Sorry.

Or at least in cases like this one, you could just return "revision" given that we have a named return value which just happens (yeah, right!) to have the correct value.

File: internal/clients/store/store.go [hunk 2]
```diff
@@ -201,36 +201,28 @@ func (store *Store) disconnected(ctx context.Context) error {
...
+	}
 
-		return nil
-	})
+	storeRoot.DefaultTimeoutConnect = time.Duration(cfg.Store.ConnectTimeout) * time.Second
+	storeRoot.DefaultTimeoutRequest = time.Duration(cfg.Store.RequestTimeout) * time.Second
+	storeRoot.DefaultTraceFlags = TraceFlags(cfg.Store.TraceLevel)
+	storeRoot.DefaultNamespaceSuffix = ""
+
+	tracing.Infof(ctx, "EP: %v TimeoutConnect: %v TimeoutRequest: %v DefTrcFlags: %v NsSuffix: %v", storeRoot.DefaultEndpoints, storeRoot.DefaultTimeoutConnect, storeRoot.DefaultTimeoutRequest, storeRoot.DefaultTraceFlags, storeRoot.DefaultNamespaceSuffix)
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply internal/clients/store/store.go:217]: Automated reformatting.  Let's deal with long lines as we decide to

File: internal/services/frontend/stepper.go [hunk 3]
```diff
@@ -69,28 +66,25 @@ func handleGetStatus(w http.ResponseWriter, r *http.Request) {
...
+	httpErrorIf(ctx, w, err)
 }
 
 // Process an http request to advance the simulated time by a specified number
 // of ticks.  The number defaults to 1.
 func handleAdvance(w http.ResponseWriter, r *http.Request) {
 	ctx, span := tracing.StartSpan(context.Background(),
 		tracing.WithName("Advance Simulated Time"),
-		tracing.WithKind(trace.SpanKindServer))
+		tracing.WithContextValue(clients.EnsureTickInContext))
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply internal/services/frontend/stepper.go:77]: Yeah, it was that the default was server, there was one client, which was removed.  All others were via AsInternal.  If we have a need to bring it back, we can.

File: internal/services/frontend/inventory.go [hunk 4]
```diff
@@ -74,11 +72,8 @@ func handlerRacksList(w http.ResponseWriter, r *http.Request) {
 
 	tracing.Infof(
 		ctx,
-		tick,
 		"Listing all %d racks, max blades/rack=%d, max blade capacity=%v",
-		rackCount,
-		res.MaxBladeCount,
-		res.MaxCapacity)
+		rackCount, res.MaxBladeCount, res.MaxCapacity)
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply internal/services/frontend/inventory.go]: Autoformatting.  But fixed.

File: internal/clients/store/store.go [hunk 5]
```diff
@@ -873,93 +881,89 @@ func checkConditions(stm concurrency.STM, req *Request) error {
 // prefix.
 //
 func (store *Store) ListWithPrefix(ctx context.Context, keyPrefix string) (response *Response, err error) {
```
[the remark this answers is not in the document]
Comment (Jim3Things) [reply internal/clients/store/store.go:893]: I'll let you tackle these.  I was trying to stay away from deep changes (yeah, sounds ironic, I know).

File: internal/clients/store/storeapi.go (hunk 1 above)
Comment (Jim3Things) [reply internal/clients/store/storeapi.go]: No worries.  Fixed.

File: internal/clients/store/store.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (MikeCJohnsonUk) [reply internal/clients/store/store.go]: Ok

File: internal/services/frontend/stepper.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (MikeCJohnsonUk) [reply internal/services/frontend/stepper.go:77]: Ok. Just wanted to make sure it wasn't inadvertently dropped.

File: internal/services/frontend/inventory.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (MikeCJohnsonUk) [reply internal/services/frontend/inventory.go]: Ok.

File: internal/clients/store/store.go (hunk 5 above)
[the remark this answers is not in the document]
Comment (MikeCJohnsonUk) [reply internal/clients/store/store.go:893]: Ok.

File: internal/clients/store/storeapi.go (hunk 1 above)
Comment (MikeCJohnsonUk) [reply internal/clients/store/storeapi.go]: Thank you.

**Review by MikeCJohnsonUk — APPROVED** [review]

**Pull request merged**